rust/operator-binary/src/discovery.rs

```diff
@@ -19,75 +17,38 @@ pub enum Error {
     },
     #[snafu(display("chroot path {} was relative (must be absolute)", chroot))]
     RelativeChroot { chroot: String },
-    #[snafu(display("object has no name associated"))]
-    NoName,
-    #[snafu(display("object has no namespace associated"))]
-    NoNamespace,
     #[snafu(display("failed to list expected pods"))]
     ExpectedPods {
         source: stackable_hive_crd::NoNamespaceError,
     },
-    #[snafu(display("could not find service port with name {}", port_name))]
-    NoServicePort { port_name: String },
-    #[snafu(display("service port with name {} does not have a nodePort", port_name))]
-    NoNodePort { port_name: String },
-    #[snafu(display("could not find Endpoints for {}", svc))]
-    FindEndpoints {
-        source: stackable_operator::error::Error,
-        svc: ObjectRef<Service>,
-    },
-    #[snafu(display("nodePort was out of range"))]
-    InvalidNodePort { source: TryFromIntError },
-    #[snafu(display("failed to build ConfigMap"))]
-    BuildConfigMap {
+    #[snafu(display("operator framework reported error"))]
+    OperatorFramework {
```

Comment:
I'd rather not do this kind of catch-all unless the errors from the framework are explanatory enough that they can actually stand on their own.

Reply:
I get that, but in this case the original error would be a kube-rs error i assume, how would i catch that?

Reply:
I don't think we need to drill down into what exactly went wrong (that's `source`'s responsibility, after all), but rather try to give some context around what we were trying to do when the error occurred (in this case, `FindEndpoints` or preparing the ConfigMap).

Reply:
done